Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -36,38 +36,6 @@ public enum DependencyResolverError: Error, Equatable, CustomStringConvertible {
 public protocol DependencyResolverDelegate {
 }
 
-/// A bound version for a package within an assignment.
```

Comment:
I think I'd slightly prefer keeping this together with `DependencyResolver` -- since it's only a small enum.

Reply:
Although it's a small enum (~40 loc), `BoundVersion` is referenced from the `DependencyResolver.Binding` typealias, the `getUpdatedIdentifier(at:)` method in `BasePackageContainer` and its subclasses, and a helper method in `Workspace`. At least for me, I find it helpful to have any public types that are referenced from more than one context to have their own file. But I can certainly revert this commit if you feel strongly about this change — just let me know.

Reply:
I don't have any strong feelings about this, unless someone else does, this seems good to merge. Thanks!

Reply:
@abertelrud Any thoughts?

Reply:
No strong opinions on this.  If it's public and referenced from multiple places, it makes sense to split it out.  Small "helper" enums that are just auxiliary to a main public type make sense to keep together with the type, even if they're public (e.g. a specialized error type thrown only by the main type), but in this case I think it makes sense to split it out.

Reply:
Thanks for weighing in, @abertelrud. I think we have strong alignment on this, so I'm going to go ahead and merge this.